src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/TaintedDataAnalysis/SourceInfo.cs

```diff
@@ -14,15 +14,16 @@ internal class SourceInfo : ITaintedDataInfo, IEquatable<SourceInfo>
         /// Constructs.
         /// </summary>
         /// <param name="fullTypeName">Full type name of the...type (namespace + type).</param>
-        /// 
         /// <param name="taintedProperties">Properties that generate tainted data.</param>
         /// <param name="taintedMethods">Methods that generate tainted data.</param>
-        public SourceInfo(string fullTypeName, bool isInterface, ImmutableHashSet<string> taintedProperties, ImmutableHashSet<string> taintedMethods)
+        /// <param name="taintConstantArray"></param>
+        public SourceInfo(string fullTypeName, bool isInterface, ImmutableHashSet<string> taintedProperties, ImmutableDictionary<string, ImmutableDictionary<string, ArgumentCheck>> taintedMethods, bool taintConstantArray)
```

Comment:
>ImmutableDictionary<string, ArgumentCheck> [](start = 137, length = 42)

Instead of a dictionary of individual argument value callbacks, another approach is to have a delegate similar to `ConstructorMapper.ValueContentAbstractValueCallback`:
`delegate bool IsInvocationTainted(IMethodSymbol, IReadOnlyList<ValueContentAbstractValue> argumentValueContents)`

I think that'd be clearer than callbacks for individual arguments.  For cases with multiple arguments, `IsInvocationTainted` could specify how the arguments relate to each other (do all arguments have to certain values to be tainted, or just one argument, or something more complicated?).  Whereas with the current approach, how arguments relate to each other is hardcoded in `IsSourceMethod` (any argument not matching `ArgumentCheck` means not tainted).

Also, I think it's harder to document how the current approach works.